solr/core/src/test/org/apache/solr/search/TestSmileRequest.java

```diff
@@ -71,7 +71,7 @@ public void testDistribJsonRequest() throws Exception {
           @Override
           public void assertJQ(SolrClient client, SolrParams args, String... tests)
               throws Exception {
-            ((HttpSolrClient) client).setParser(SmileResponseParser.inst);
+            ((HttpSolrClient) client).setParser(new SmileResponseParser());
```

Comment:
What issues are you running into with this test?

Reply:
I can't figure out how to move away from the `.setParser`, I want to somehow override how `SolrTestCaseHS` sets up it's `SolrClient` and kind of got burned out trying to slueth thorugh all the layers of indirection...

Reply:
Okay but surely moving away from a global instance to creating a new instance (of this parser thing) will have zero effect on solving these complications around SolrClient lifecycle/mutability.

Looking at this specific scenario, it seems to me we need/want to be able to set the parser for a specific request, which will not run afoul of our immutability plans for SolrClient.  At the SolrClient level, setting this parser is more global (across requests).  SolrQueryRequest has a setResponseParser; try that.

Reply:
Thanks for the tip, that worked great!